Networking/NetworkingTests/Remote/GenerativeContentRemoteTests.swift

```diff
@@ -226,6 +226,287 @@ final class GenerativeContentRemoteTests: XCTestCase {
+        let prompt = try XCTUnwrap(request.parameters?["prompt"] as? String)
+        XCTAssertTrue(prompt.contains("tags:"))
+        XCTAssertTrue(prompt.contains("Food, Grocery"))
+    }
+
+    func test_generateProduct_prompt_does_not_have_tags_if_no_tags_available() async throws {
+        // Given
+        let remote = GenerativeContentRemote(network: network)
+        network.simulateResponse(requestUrlSuffix: "sites/\(sampleSiteID)/jetpack-openai-query/jwt", filename: "jwt-token-success")
+        network.simulateResponse(requestUrlSuffix: "text-completion", filename: "generate-product-success")
+
+        // When
+        _ = try await remote.generateProduct(siteID: sampleSiteID,
+                                             productName: "Cookie",
+                                             keywords: "Crunchy, Crispy",
+                                             language: "en",
+                                             tone: "Casual",
+                                             currencySymbol: "INR",
+                                             dimensionUnit: "cm",
+                                             weightUnit: "kg",
+                                             categories: [],
+                                             tags: [])
+
+        // Then
+        let request = try XCTUnwrap(network.requestsForResponseData.last as? DotcomRequest)
+        let prompt = try XCTUnwrap(request.parameters?["prompt"] as? String)
+        XCTAssertFalse(prompt.contains("tags:"))
+        XCTAssertFalse(prompt.contains("Food, Grocery"))
+    }
+
+    func test_generateProduct_with_success_returns_product() async throws {
+        // Given
+        let remote = GenerativeContentRemote(network: network)
+        network.simulateResponse(requestUrlSuffix: "sites/\(sampleSiteID)/jetpack-openai-query/jwt", filename: "jwt-token-success")
+        network.simulateResponse(requestUrlSuffix: "text-completion", filename: "generate-product-success")
+
+        // When
+        let product = try await remote.generateProduct(siteID: sampleSiteID,
+                                                        productName: "Cookie",
+                                                        keywords: "Crunchy, Crispy",
+                                                        language: "en",
+                                                        tone: "Casual",
+                                                        currencySymbol: "INR",
+                                                        dimensionUnit: "cm",
+                                                        weightUnit: "kg",
+                                                        categories: [ProductCategory.fake(), ProductCategory.fake()],
+                                                        tags: [ProductTag.fake(), ProductTag.fake()])
+
+        // Then
+        XCTAssertNotNil(product)
```

Comment:
Nit: Can we check and compare product details too?

Reply:
Done in 0e1402521d415ff4387a0711269114856f92ae82